core.h

```diff
@@ -21,8 +21,29 @@
 #pragma once
 
 #define _GNU_SOURCE
+#include <stdbool.h>
 #include <dlfcn.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
 
 #define __str(s) #s
 #define _str(s) __str(s)
 #define NEXT(name) ((typeof(name) *) dlsym(RTLD_NEXT, _str(name)))
+
+static inline bool
+is_tls_domain(int domain)
+{
+    return domain == AF_INET || domain == AF_INET6;
+}
+
+static inline bool
+is_tls_type(int type)
+{
+    return type == SOCK_STREAM || type == SOCK_DGRAM;
+}
+
+static inline bool
+is_tls_protocol(int protocol)
+{
+    return protocol == 0 || protocol == IPPROTO_TCP || protocol == IPPROTO_UDP;
+}
```

Comment:
I didn't think this through. I don't want the socket headers polluting `core.h`. So let's make the helpers full functions. You'll need to create core.c and add it to the build system.